tests/metrics/metric_manager/test_metric_manager.py

```diff
@@ -0,0 +1,152 @@
+
+from .data_generator import FakeData, FakeDataGenerator
+
+
+@METRICS.register_class
+class MocSumMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('lol_sum', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        self.lol_sum += 1
+
+    def compute(self):
+        return self.lol_sum
+
+
+@METRICS.register_class
+class MockSumMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('sum', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        self.sum += 1
+
+    def compute(self):
+        return self.sum
+
+
+@METRICS.register_class
+class MockConstantMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.add_state('constant', default=torch.tensor(0), dist_reduce_fx=None)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        return
+
+    def compute(self):
+        return self.constant
+
+
+@METRICS.register_class
+class MocRaiseMetric(Metric):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def update(self, predict: torch.Tensor, target: torch.Tensor):
+        return
+
+    def compute(self, memory_block):
+        return torch.tensor([1, 2])
+
+
+def run_metric_manager(class_names: List[str], names: List[str], \
+                       target_fields: List[Dict], data_generator: FakeDataGenerator):
+    if len(class_names) != len(names) or len(names) != len(target_fields):
+        raise print('Not correct Test! Input params not same length.')
+
+    metric_params = []
+    for i in range(len(class_names)):
+        params = MetricParams(name=class_names[i], mapping=target_fields[i], log_name=names[i])
+        metric_params.append(params)
+
+    metric_manager = MetricManager(metric_params)
+    print(f'type = {type(metric_manager)}')
+    for i in range(len(data_generator)):
+        metric_manager(Phase.TRAIN, **data_generator[i])
+
+    return metric_manager.on_epoch_end(Phase.TRAIN)
+
+class TestCase:
+    def __init__(self, test_name: str, class_names: List[str], names: List[str], \
+                target_fields: List[Dict], data_generator, expected):
+        self.test_name = test_name
+        self.class_names = class_names
+        self.names = names
+        self.target_fields = target_fields
+        self.data_generator = data_generator
+        self.expected = expected
+
+
+# generate data
+embedding_data = FakeData(name='embedding', shape = [512], num_repeats=5)
+target_data = FakeData(name='target', shape=[10], num_repeats=5)
+embedding_fake_data = [embedding_data, target_data]
+data_generator = FakeDataGenerator(embedding_fake_data)
+target_fields = dict(predict='embedding', target='target')
+
+
+class MetricManagerTest(unittest.TestCase):
+    def test(self):
```

Comment:
Rename according to template: `test_something_when_something_happens`

Reply:
Lol, I meant to replaces <something> and <something_happens>. Check my example: `def test_no_weight_decay_in_optimizers_when_module_contains_1d_tensor(self):`